cpp/tests/groupby/lists_tests.cu

```diff
@@ -0,0 +1,200 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "groupby_test_util.hpp"
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/iterator_utilities.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <cudf/concatenate.hpp>
+#include <cudf/detail/aggregation/aggregation.hpp>
+#include <cudf/table/experimental/row_operators.cuh>
+#include <cudf/table/table_view.hpp>
+#include <cudf/types.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/logical.h>
+
+#include <vector>
+
+namespace cudf {
+namespace test {
+
+template <typename V>
+struct groupby_lists_test : public cudf::test::BaseFixture {
+};
+
+TYPED_TEST_SUITE(groupby_lists_test, cudf::test::FixedWidthTypes);
+
+using namespace cudf::test::iterators;
+
+using R = cudf::detail::target_type_t<int32_t, aggregation::SUM>;  // Type of aggregation result.
+using strings = strings_column_wrapper;
+using structs = structs_column_wrapper;
+
+template <typename T>
+using fwcw = cudf::test::fixed_width_column_wrapper<T>;
+
+template <typename T>
+using lcw = cudf::test::lists_column_wrapper<T, int32_t>;
+
+namespace {
+static constexpr auto null = -1;
+
+// Checking with a single aggregation, and aggregation column.
+// This test is orthogonal to the aggregation type; it focuses on testing the grouping
+// with LISTS keys.
+auto sum_agg() { return cudf::make_sum_aggregation<groupby_aggregation>(); }
+
+// TODO: this is a naive way to compare expected key/value against resulting key/value. To be
+// replaced once list lex comparator is supported (https://github.com/rapidsai/cudf/issues/5890)
+template <typename Equal>
+struct match_expected_fn {
+  match_expected_fn(cudf::size_type const num_rows, Equal equal)
+    : _num_rows{num_rows}, _equal{equal}
+  {
+  }
+
+  __device__ bool operator()(cudf::size_type const idx)
+  {
+    for (auto i = _num_rows; i < 2 * _num_rows; i++) {
+      if (_equal(idx, i)) { return true; }
+    }
+    return false;
+  }
+
+  cudf::size_type const _num_rows;
+  Equal _equal;
+};
+
+inline void test_hash_based_sum_agg(column_view const& keys,
+                                    column_view const& values,
+                                    column_view const& expect_keys,
+                                    column_view const& expect_vals)
+{
+  auto agg                     = cudf::make_sum_aggregation<groupby_aggregation>();
+  auto const include_null_keys = null_policy::INCLUDE;
+  auto const keys_are_sorted   = sorted::NO;
+
+  std::vector<groupby::aggregation_request> requests;
+  requests.emplace_back(groupby::aggregation_request());
+  requests[0].values = values;
+
+  requests[0].aggregations.push_back(std::move(agg));
+
```

Comment:
Why do this instead of `requests.emplace_back(values, cudf::make_sum_aggregation<groupby_aggregation>())`?

Reply:
push_back should work.

If you want to use `emplace_back`, instead of accessing the just added value using subscript, you can do 
```c++
auto& request = requests.emplace_back(groupby::aggregation_request());
request.values=values;
request.aggregations.push_back(std::move(agg));
```